zcash_primitives/src/transaction/builder.rs

```diff
@@ -377,7 +423,8 @@ impl<R: RngCore + CryptoRng> Builder<R> {
         value: Amount,
         memo: Option<Memo>,
     ) -> Result<(), Error> {
-        let output = SaplingOutput::new(&mut self.rng, ovk, to, value, memo)?;
+        let output =
+            SaplingOutput::new(self.mtx.expiry_height, &mut self.rng, ovk, to, value, memo)?;
```

Comment:
(look at line 355)
define `height` field in `Builder` struct, and compute `expiry_height` in `Builder`